src/components/Icon/index.js

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 function Icon({
   className,
   family,
   type,
+  hoverIconType,
   ...rest
 }) {
-  function getCSSClasses() {
+  const [isHovered, setIsHovered] = useState(false);
+  function getCSSClasses(showHoverIcon) {
     const fontFamily = family || 'fa';
 
     return [
       'tyk-icon',
       fontFamily,
-      `${fontFamily}-${type}`,
+      `${fontFamily}-${showHoverIcon ? hoverIconType : type}`,
       className,
     ].filter(Boolean).join(' ');
   }
 
-  return <i className={getCSSClasses()} {...rest} />;
+  const toggleHover = () => setIsHovered(!isHovered);
+
+  return (
+    <span onMouseEnter={toggleHover} onMouseLeave={toggleHover}>
```

Comment:
Is there a reason why we add an additional tag for the component instead of adding the events directly on the `i` tag?
```suggestion
    return (
      <i
        className={getCSSClasses()}
        onMouseEnter={toggleHover}
        onMouseLeave={toggleHover}
        {...rest}
      />;
```